src/client/components/vision/camera/view_model.ts

```diff
@@ -93,9 +97,40 @@ export class CameraViewModel {
     if (this.model.visualmesh) {
       scene.add(this.visualmesh(this.model.visualmesh))
     }
+    this.model.balls.forEach(ball => scene.add(this.ball(ball)))
+    this.model.goals.forEach(goal => scene.add(this.goal(goal)))
     return scene
   }
 
+  private ball = createTransformer((m: Ball) => {
+    const Hwc = new Matrix4().getInverse(toThreeMatrix4(m.Hcw))
+    const imageHcw = this.model.image ? toThreeMatrix4(this.model.image.Hcw) : new Matrix4()
+    const Hcc = imageHcw.multiply(Hwc)
+    return this.makeCone({
+      axis: toThreeVector3(m.cone.axis).applyMatrix4(Hcc),
+      gradient: m.cone.gradient,
+      colour: new Vector4(0, 0, 1, 0.7),
+      lineWidth: 10,
+    })
+  })
+
+  private goal = createTransformer((m: Goal) => {
+    const goal = new Object3D()
+    const Hwc = new Matrix4().getInverse(toThreeMatrix4(m.Hcw))
+    const imageHcw = this.model.image ? toThreeMatrix4(this.model.image.Hcw) : new Matrix4()
+    const Hcc = imageHcw.multiply(Hwc)
+    const tl = toThreeVector3(m.frustum.tl).applyMatrix4(Hcc)
+    const tr = toThreeVector3(m.frustum.tr).applyMatrix4(Hcc)
+    const bl = toThreeVector3(m.frustum.bl).applyMatrix4(Hcc)
+    const br = toThreeVector3(m.frustum.br).applyMatrix4(Hcc)
+    const colour = new Vector4(0.8, 0.8, 0, 0.5)
+    goal.add(this.makePlaneSegment({ start: tl, end: bl, colour, lineWidth: 10 }))
```

Comment:
You could also make this a map function if there is something like pythons zip :P